addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.handler;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.netatmo.config.NetatmoBridgeConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.client.ApiClient;
+import io.swagger.client.api.StationApi;
+import io.swagger.client.api.ThermostatApi;
+import retrofit.RestAdapter.LogLevel;
+
+/**
+ * {@link NetatmoBridgeHandler} is the handler for a Netatmo API and connects it
+ * to the framework. The devices and modules uses the
+ * {@link NetatmoBridgeHandler} to request informations about their status
+ *
+ * @author Gaël L'hopital - Initial contribution OH2 version
+ *
+ */
+public class NetatmoBridgeHandler extends BaseBridgeHandler {
+    private static Logger logger = LoggerFactory.getLogger(NetatmoBridgeHandler.class);
+    private NetatmoBridgeConfiguration configuration;
+    private ApiClient apiClient;
+    private StationApi stationApi;
+    private ThermostatApi thermostatApi;
+
+    public NetatmoBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Initializing Netatmo API bridge handler.");
+
+        this.configuration = getConfigAs(NetatmoBridgeConfiguration.class);
+
+        this.apiClient = new ApiClient("password_oauth", configuration.clientId, configuration.clientSecret,
+                configuration.username, configuration.password);
+
+        apiClient.getTokenEndPoint().setScope(getApiScope(configuration));
+        apiClient.getAdapterBuilder().setLogLevel(LogLevel.FULL);
+
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
Are you sure at that point that the credentials are ok and that the server is reachable? Only should set the Bridge to ONLINE only upon successful connection - otherwise you need to set it to OFFLINE with an according message, what is wrong.